packages/server/socketHelpers/sendEncodedMessage.ts

```diff
@@ -1,16 +1,57 @@
 import {HttpResponse, WebSocket} from 'uWebSockets.js'
+import handleDisconnect from '../socketHandlers/handleDisconnect'
 import sendSSEMessage from '../sse/sendSSEMessage'
+import ConnectionContext from './ConnectionContext'
 import isHttpResponse from './isHttpResponse'
 
 const ESTIMATED_MTU = 1400
-const sendEncodedMessage = (transport: WebSocket | HttpResponse, message: object | string) => {
-  if (transport.done) return
-  const str = JSON.stringify(message)
-  if (isHttpResponse(transport)) {
-    sendSSEMessage(transport as HttpResponse, str)
+const TIMEOUT_COEFFICIENT = 2.0
+const INITIAL_TIMEOUT = 2000 // 2 seconds
+const MAX_TIMEOUT = 10000 // 10 seconds
+const TIMEOUT_JITTER = 20 // 20 ms
+
+const sendAndPushToReliableQueue = (
+  context: ConnectionContext,
+  mid: number,
+  object: any,
+  timeout = INITIAL_TIMEOUT + Math.random() * TIMEOUT_JITTER
+) => {
+  const {socket, reliableQueue} = context
+  if (timeout > MAX_TIMEOUT) {
+    handleDisconnect(context)
     return
   }
-  transport.send(str, false, str.length > ESTIMATED_MTU)
+  const message = JSON.stringify([object, mid])
+  sendEncodedMessageBasedOnSocket(socket, message)
+  const timer = setTimeout(() => {
```

Comment:
we've already got a client where it asks for messages it missed (pull queue).
if the server also resends messages (push queue) then that means it's possible that the server could send the same message twice even if the client didn't request it.
that means trebuchet-client can't have just a `requestedMids` array, but it'll need to keep track of EVERY message that comes in, cause any message could be a duplicate! 

as a first pass, i'd suggest we just get rid of the `setTimeout` on the server. it'll reduce complexity & give us about the same quality, although i could be convinced either way